src/SumUp/HttpClients/Response.php

```diff
@@ -95,11 +95,25 @@ protected function parseResponseForErrors()
             throw new SumUpValidationException($invalidFields, $this->httpResponseCode);
         }
         if ($this->httpResponseCode >= 500) {
-            $message = (!is_null($this->body) && !is_null($this->body->message)) ? $this->body->message : $this->body;
+            $message = 'Server error';
+            if (!is_null($this->body)) {
+                if (!is_null($this->body->message)) {
+                    $message = $this->body->message;
+                } elseif (!is_null($this->body->error_message)) {
+                    $message = $this->body->error_message;
+                }
+            }
             throw new SumUpServerException($message, $this->httpResponseCode);
         }
         if ($this->httpResponseCode >= 400) {
-            $message = (!is_null($this->body) && !is_null($this->body->message)) ? $this->body->message : $this->body;
+            $message = 'Error';
+            if (!is_null($this->body)) {
+                if (!is_null($this->body->message)) {
+                    $message = $this->body->message;
+                } elseif (!is_null($this->body->error_message)) {
+                    $message = $this->body->error_message;
+                }
+            }
```

Comment:
I see that the error parsing code is the same so it should perhaps be in its own function.

If 500+ errors are server errors, then 400+ errors should be client errors.

I would try to escape nesting the `if`s, e.g.:
```php
$message = 'Client error';

if (is_null($this->body)) {
    return $message;
}

$message = $this->body->message ?: $message;
$message = $this->body->error_message ?: $message;

return $message;
```